mountpoint-s3/src/inode/readdir.rs

```diff
@@ -456,13 +456,16 @@ mod ordered {
                 // Deduplicate the entry we want to return
                 match next {
                     Some(entry) => {
-                        if self.last_name.as_deref() == Some(entry.name()) {
+                        let last_name = self.last_entry.as_ref().map(|entry| entry.name());
+                        let last_description = self.last_entry.as_ref().map(|entry| entry.description());
+                        if last_name == Some(entry.name()) {
```

Comment:
Probably changing this check to `match &self.last_entry` would be clearer, so we know that if last_entry exists then last_name and last_description must have some values.